core/domain/exp_services.py

```diff
@@ -487,7 +487,14 @@ def apply_change_list(exploration_id, change_list):
             elif change.cmd == exp_domain.CMD_ADD_TRANSLATION:
                 exploration.states[change.state_name].add_translation(
                     change.content_id, change.language_code,
-                    change.translation_html)
+                    change.translation_html, change.data_format,
+                    change.needs_update)
+            elif (change.cmd ==
+                  exp_domain.CMD_MARK_TRANSLATION_AS_NEEDING_UPDATE):
+                exploration.states[
+                    change.state_name
+                ].written_translations.mark_translation_as_needing_update(
+                    change.content_id)
```

Comment:
Generally, we do not expose the internal properties of domain objects this much in the service layer. We are exposing the state here because the commands are state-specific. Can we have  `mark_translation_as_needing_update` as a function for state object similar to `add_translation`?